File: packages/share_movie_plugin/ios/Classes/ShareMoviePlugin.m
#import "ShareMoviePlugin.h"
#if __has_include(<share_movie_plugin/share_movie_plugin-Swift.h>)
#import <share_movie_plugin/share_movie_plugin-Swift.h>
#else
#import "share_movie_plugin-Swift.h"
#endif

@implementation ShareMoviePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftShareMoviePlugin registerWithRegistrar:registrar];
}
@end

File: packages/share_movie_plugin/macos/Classes/ShareMoviePlugin.swift
import Cocoa
import FlutterMacOS

public class ShareMoviePlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "share_movie_plugin", binaryMessenger: registrar.messenger)
        let instance = ShareMoviePlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if call.method == "shareMovie" {
            let arguments = call.arguments as! [String: Any]
            let message = arguments["message"] as! String
            shareMovie(message)
            result(nil)
        } else {
            result(FlutterMethodNotImplemented)
        }
    }
    
    private func shareMovie(_ message: String) {
        let picker = NSSharingServicePicker(items: [message])
        
        DispatchQueue.main.async {
            if let contentView = NSApplication.shared.keyWindow?.contentView {
                picker.show(relativeTo: .zero, of: contentView, preferredEdge: .minY)
            }
        }
    }
}
